lib/Sema/CSDiag.cpp

```diff
@@ -1325,7 +1397,8 @@ void CalleeCandidateInfo::filterList(ArrayRef<CallArgParam> actualArgs) {
     // If this isn't a function or isn't valid at this uncurry level, treat it
     // as a general mismatch.
     if (!inputType) return { CC_GeneralMismatch, {}};
-    return evaluateCloseness(inputType, actualArgs);
+    Decl *decl = candidate.getDecl();
+    return evaluateCloseness(decl ? decl->getInnermostDeclContext() : nullptr, inputType, actualArgs);
```

Comment:
Last nitpick before I merge this -- this line is too long, wrap it after the nullptr